src/FormControl/FormControl.tsx

```diff
@@ -55,49 +43,17 @@ const FormControl = React.forwardRef<HTMLDivElement, FormControlProps>(
       leadingVisual: FormControlLeadingVisual,
       validation: FormControlValidation,
     })
-    const expectedInputComponents = [
-      Autocomplete,
-      Checkbox,
-      Radio,
-      Select,
-      TextInput,
-      TextInputWithTokens,
-      Textarea,
-      InlineAutocomplete,
-    ]
+
     const choiceGroupContext = useContext(CheckboxOrRadioGroupContext)
     const disabled = choiceGroupContext.disabled || disabledProp
     const id = useSSRSafeId(idProp)
     const validationMessageId = slots.validation ? `${id}-validationMessage` : undefined
     const captionId = slots.caption ? `${id}-caption` : undefined
     const validationStatus = slots.validation?.props.variant
-    const InputComponent = childrenWithoutSlots.find(child =>
-      expectedInputComponents.some(inputComponent => React.isValidElement(child) && child.type === inputComponent),
-    )
-    const inputProps = React.isValidElement(InputComponent) && InputComponent.props
-    const isChoiceInput =
-      React.isValidElement(InputComponent) && (InputComponent.type === Checkbox || InputComponent.type === Radio)
 
-    if (InputComponent) {
-      if (inputProps?.id) {
-        // eslint-disable-next-line no-console
-        console.warn(
-          `instead of passing the 'id' prop directly to the input component, it should be passed to the parent component, <FormControl>`,
-        )
-      }
-      if (inputProps?.disabled) {
-        // eslint-disable-next-line no-console
-        console.warn(
-          `instead of passing the 'disabled' prop directly to the input component, it should be passed to the parent component, <FormControl>`,
-        )
-      }
-      if (inputProps?.required) {
-        // eslint-disable-next-line no-console
-        console.warn(
-          `instead of passing the 'required' prop directly to the input component, it should be passed to the parent component, <FormControl>`,
-        )
-      }
-    }
+    const isChoiceInput = childrenWithoutSlots.some(
+      el => React.isValidElement(el) && (el.type === Checkbox || el.type === Radio),
```

Comment:
There's no good way to detect if the wrapped component is a `Checkbox` or `Radio` without referencing them directly. I don't like this since it means we still have to have this backwards data flow from children to parent, but I don't have an elegant solution to that problem.

Reply:
Is the slot info something that could be passed along in the context value, potentially? Or does that not really work?

Reply:
I just tried this and honestly I don't think it makes it any cleaner. We still have to look through the children to figure out if it's a choice input, so all that changes is that we pass the slots info down. I think since these checks don't reference `props`, it makes sense to leave them here for now.